src/igr-tests/igr-runner.cc

```diff
@@ -47,52 +45,88 @@ void cycles_test();
 
 int main(int argc, char **argv)
 {
-    void (*test_funcs[])() = { basic_test, environ_test, environ2_test, ps_environ_test,
-            chain_to_test, force_stop_test, restart_test, check_basic_test, check_cycle_test,
-            check_cycle2_test, check_lint_test, reload1_test, reload2_test, no_command_error_test,
-            add_rm_dep_test, var_subst_test, svc_start_fail_test, dep_not_found_test,
-            pseudo_cycle_test, before_after_test, before_after2_test, log_via_pipe_test,
-            catlog_test, offline_enable_test, xdg_config_test, cycles_test };
+    struct test tests[] = { { "basic", basic_test }, { "environ", environ_test },
```

Comment:
> please can you make this kind of change separately in future (at least as a separate commit). The change isn't necessary at all for the main purpose of the PR, so it should be separate. (Doesn't need to be a separate PR since it's minor, but it should definitely be a separate commit).

I think it is, It's my solution for running individual test passed by meson tests wrapper, Also it's the main thing in my first commit: https://github.com/davmac314/dinit/pull/344/commits/542089410916d830b1e38500f5909b5bfa4267d7

> `test_dirs` is still used in code, The purpose of this PR is not to remove old infrastructure (Maybe It should?)

Reply:
>  Also it's the main thing in my first commit

You're right, it is mostly separated from the rest of the PR. It's still a refactoring though and wasn't really necessary for the change in that commit, which could also have been done without the refactoring. I'd rather keep refactoring in separate commits. It's not a huge deal for a case like this, just something to keep in mind.

Reply:
> test_dirs is still used in code, The purpose of this PR is not to remove old infrastructure (Maybe It should?)

There's no point having both a list of test structs which have the test name, and a separate list of test names. The PR didn't need to introduce the struct at all, but seeing as it does, then the old separate list should also be removed, yes. (You could have made it work without introducing the structure by just searching the test names array to find the correct index, and using that index to find the function from the test functions array).

I think the change of combing two arrays (test function, test name) into a single array (of struct with both function and name) is a good change - but it's not good to duplicate information (such as test names) so the old test-names array should go.

Reply:
It's not resolved, the old array (`test_dirs`) still is there.

Reply:
> There's no point having both a list of test structs which have the test name, and a separate list of test names. The PR didn't need to introduce the struct at all, but seeing as it does, then the old separate list should also be removed, yes. (You could have made it work without introducing the structure by just searching the test names array to find the correct index, and using that index to find the function from the test functions array).

I didn't think about it, And yeah but it's a little hacky.

> I think the change of combing two arrays (test function, test name) into a single array (of struct with both function and name) is a good change - but it's not good to duplicate information (such as test names) so the old test-names array should go.

So I will add a commit to remove old infrastructure.

Reply:
Ideally, squash it into that first commit (as a fixup). Or I can do that when merging this.

Reply:
Check my commits and resolve this if everything is ok